sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -104,36 +105,52 @@ def test_cdist(metric_param_grid, X, Y):
 
         dm = DistanceMetricInterface.get_metric(metric, **kwargs)
         D_sklearn = dm.pairwise(X, Y)
-        assert_allclose(D_sklearn, D_scipy_cdist)
+        assert_allclose(D_sklearn, D_scipy_cdist, **rtol_dict)
+
+        D_sklearn_csr = dm.pairwise(X_csr, Y_csr)
+        assert_allclose(D_sklearn_csr, D_scipy_cdist, **rtol_dict)
 
 
 @pytest.mark.parametrize("metric", BOOL_METRICS)
 @pytest.mark.parametrize(
     "X_bool, Y_bool", [(X_bool, Y_bool), (X_bool_mmap, Y_bool_mmap)]
 )
 def test_cdist_bool_metric(metric, X_bool, Y_bool):
-    D_true = cdist(X_bool, Y_bool, metric)
+    D_scipy_cdist = cdist(X_bool, Y_bool, metric)
+
+    if metric == "jaccard" and _IS_32BIT:
+        pytest.skip("Jaccard Distance on 32bit architecture is unstable.")
```

Comment:
This is the skip for the 32bit architecture.

Reply:
I think would rather `xfail` instead of skip.

But I would not block the merge of this PR for such a problem which is unlikely to impact many users in practice.